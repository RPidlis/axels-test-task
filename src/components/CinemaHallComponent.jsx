import { Container, Row } from 'react-bootstrap';

import { Seat } from '../styled/components/cinemaHall';

const CinemaHallComponent = ({ sessionSeats,seats }) => (
  <Container className="d-flex flex-column justify-content-around p-0">
    <Row className="d-flex justify-content-between pb-2">
      {seats.map((item) => (
        <Seat
          key={item.id}
          variant={sessionSeats.includes(item.id) ? 'info' : 'outline-info'}
          disabled={sessionSeats.includes(item.id)}
        >
          {item.id}
        </Seat>
      ))}
    </Row>
  </Container>
);

export default CinemaHallComponent;
