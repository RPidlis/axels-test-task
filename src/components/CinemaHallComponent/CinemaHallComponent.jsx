import styled from 'styled-components'

const CinemaHall = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  width: 400px;
  height: 200px;
`
const CinemaRow = styled.div`
  display: flex;
  justify-content: space-around;
`
const Seat = styled.div`
  width: 30px;
  height: 40px;
  border-radius: 7px;
  background-color: red;
`


const CinemaHallComponent = () => {
    return (
        <CinemaHall>
            <CinemaRow>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
            </CinemaRow>
            <CinemaRow>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
            </CinemaRow>
            <CinemaRow>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
                <Seat/>
            </CinemaRow>
        </CinemaHall>
    )
}

export default CinemaHallComponent
