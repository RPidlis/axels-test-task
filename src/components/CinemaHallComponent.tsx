import { Container, Row } from 'react-bootstrap';
import React from 'react';

import { Seat } from '../styled/components/cinemaHall';
import { SeatType } from '../redux/ducks/schedule';

export type PropsType = {
  sessionSeats: Array<number> | null;
  seats: Array<SeatType> | null;
  chosenSeats: Array<number>;
  onSeatClick: (id: number) => void;
};

const CinemaHallComponent: React.FC<PropsType> = ({
  sessionSeats,
  seats,
  chosenSeats,
  onSeatClick,
}) => (
  <Container className="d-flex flex-column justify-content-around p-0">
    <Row className="d-flex justify-content-between pb-2">
      {seats?.map((seat) => {
        const isSeatSold = sessionSeats?.includes(seat.id);
        const isSeatChoized = chosenSeats?.includes(seat.id) || false;
        return (
          <Seat
            active={isSeatChoized}
            key={seat.id}
            variant={
              isSeatChoized ? 'info' : isSeatSold ? 'info' : 'outline-info'
            }
            disabled={isSeatSold}
            onClick={() => onSeatClick(seat.id)}
          >
            {seat.id}
          </Seat>
        );
      })}
    </Row>
  </Container>
);

export default CinemaHallComponent;
