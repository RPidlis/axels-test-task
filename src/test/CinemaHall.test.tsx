import { render } from '@testing-library/react';

import CinemaHallComponent, {
  PropsType,
} from '../components/CinemaHallComponent';

function renderCinemaHallComponent(props: Partial<PropsType> = {}) {
  const defaultProps: PropsType = {
    sessionSeats: [1, 3, 2, 4, 5],
    seats: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }, { id: 5 }, { id: 6 }],
    chosenSeats: [],
    onSeatClick() {
      return;
    },
  };
  return render(<CinemaHallComponent {...defaultProps} {...props} />);
}

describe('test renders Cinema Hall', () => {
  it('test renders snapshot ', () => {
    const component = renderCinemaHallComponent();
    expect(component).toMatchSnapshot();
  });
  it('test disables session seats', () => {
    const { getByText } = renderCinemaHallComponent();
    const button = getByText(1);
    expect(button).toBeDisabled();
  });

  it('test enables session seats', () => {
    const { getByText } = renderCinemaHallComponent();
    const button = getByText(6);
    expect(button).toBeEnabled();
  });

  it('test renders seats on cinema hall', () => {
    const { queryAllByRole } = renderCinemaHallComponent();
    const buttons = queryAllByRole('button');
    expect(buttons.length).toBe(6);
  });
});
