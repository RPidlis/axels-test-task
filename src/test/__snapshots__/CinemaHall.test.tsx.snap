// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test renders Cinema Hall test renders snapshot  1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="d-flex flex-column justify-content-around p-0 container"
      >
        <div
          class="d-flex justify-content-between pb-2 row"
        >
          <button
            class="sc-bdvvaa kUGjmq btn btn-info"
            disabled=""
            type="button"
          >
            1
          </button>
          <button
            class="sc-bdvvaa kUGjmq btn btn-info"
            disabled=""
            type="button"
          >
            2
          </button>
          <button
            class="sc-bdvvaa kUGjmq btn btn-info"
            disabled=""
            type="button"
          >
            3
          </button>
          <button
            class="sc-bdvvaa kUGjmq btn btn-info"
            disabled=""
            type="button"
          >
            4
          </button>
          <button
            class="sc-bdvvaa kUGjmq btn btn-info"
            disabled=""
            type="button"
          >
            5
          </button>
          <button
            class="sc-bdvvaa kUGjmq btn btn-outline-info"
            type="button"
          >
            6
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="d-flex flex-column justify-content-around p-0 container"
    >
      <div
        class="d-flex justify-content-between pb-2 row"
      >
        <button
          class="sc-bdvvaa kUGjmq btn btn-info"
          disabled=""
          type="button"
        >
          1
        </button>
        <button
          class="sc-bdvvaa kUGjmq btn btn-info"
          disabled=""
          type="button"
        >
          2
        </button>
        <button
          class="sc-bdvvaa kUGjmq btn btn-info"
          disabled=""
          type="button"
        >
          3
        </button>
        <button
          class="sc-bdvvaa kUGjmq btn btn-info"
          disabled=""
          type="button"
        >
          4
        </button>
        <button
          class="sc-bdvvaa kUGjmq btn btn-info"
          disabled=""
          type="button"
        >
          5
        </button>
        <button
          class="sc-bdvvaa kUGjmq btn btn-outline-info"
          type="button"
        >
          6
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
